pkg/gofr/datasource/pubsub/mqtt/mqtt.go

```diff
@@ -338,19 +342,39 @@ func (m *MQTT) Unsubscribe(topic string) error {
 		close(sub.msgs)
 		delete(m.subscriptions, topic)
 	}
-	m.mu.RUnlock()
 
 	return nil
 }
 
-func (m *MQTT) Disconnect(waitTime uint) {
+func (m *MQTT) Close() error {
+	if m.config.CloseTimeout != 0 {
+		m.Client.Disconnect(m.config.CloseTimeout)
+	}
+
+	// setting default value if value is not populated
+	m.Client.Disconnect(uint(CloseTimeout.Milliseconds()))
```

Comment:
This is for the scenario when the user has not populated the `CloseTimeout` value in Configs while creating new client using `New` method, in that case if we go to default value it will be zero, so it will skip the condition
**if m.config.CloseTimeout != 0 {**


therefore setting our constant value here

Reply:
If the user has defined the value we are picking it from config